Performance degradation when quitting on Windows 10

**Describe the bug**
Since 0.16.0 and onwards, quitting GitUI seems to be significantly slower than it used to on Windows. This does not appear to be the case on my Linux machines, however.

**To Reproduce**
Steps to reproduce the behavior:
1. Install gitui @ 0.16.2
2. Run a terminal like cmd.exe (also tested in pwsh.exe)
3. Navigate to a repository
4. Open gitui in a repository
5. Quit gitui
6. Wait 1-3 seconds (unsure if depends on hardware specs and repository)

**Expected behavior**
On 0.15.0 gitui quits almost immediately. I would expect the same on 0.16.x+.

**Screenshots**
To better illustrate, here's a GIF where I compare the two versions:
![gitui-performance](https://user-images.githubusercontent.com/3541466/126780180-81717f73-115d-4550-8bb1-7dabebdf0338.gif)

**Context (please complete the following information):**
 - OS/Distro + Version: `Windows 10 Pro 10.0.19043 Build 19043`
 - GitUI Version `0.16.2`
 - Rust version: `1.5.0`

**Additional context**
Installed through scoop. I've also tested on cargo install but the problem persists unfortunately.

@extrawurst Here you go :)

#### Software version

gitui 0.16.2 (aacc7a3-modified)

#### Operating system

Windows 6.2.9200

#### Compile time information

- Profile: release
- Target triple: x86_64-pc-windows-msvc
- Family: windows
- OS: windows
- Architecture: x86_64
- Pointer width: 64
- Endian: little
- CPU features: fxsr,sse,sse2
- Host: x86_64-pc-windows-msvc

#### Environment variables

```bash
SHELL=<not set>
EDITOR=<not set>
GIT_EDITOR=<not set>
VISUAL=<not set>
```

#### Command-line

```bash
C:\Users\patrick.christensen\scoop\apps\gitui\current\gitui.exe --bugreport
```

Log when running `v0.15.0`:
```
14:36:11 [TRACE] (1) gitui::app: [src\app.rs:305] update
14:36:11 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:95] request: [hash: 4747929034360955961] (type: WorkingDir, untracked: true)
14:36:11 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:95] request: [hash: 11643232720821160596] (type: Stage, untracked: true)
14:36:11 [TRACE] (3) asyncgit::status: [asyncgit\src\status.rs:152] status fetched: 11643232720821160596 (type: Stage, untracked: true)
14:36:11 [TRACE] (2) asyncgit::status: [asyncgit\src\status.rs:152] status fetched: 4747929034360955961 (type: WorkingDir, untracked: true)
14:36:11 [TRACE] (1) gitui::app: [src\app.rs:237] event: Input(Resize(118, 30))
14:36:11 [TRACE] (1) gitui::app: [src\app.rs:323] update_git: Status
14:36:11 [TRACE] (1) gitui::app: [src\app.rs:323] update_git: Status
14:36:12 [TRACE] (1) gitui::app: [src\app.rs:237] event: Input(Key(KeyEvent { code: Char('c'), modifiers: CONTROL }))
```

Log when running `v.0.16.2` (seems to throw an error at the end there 🤔):
```
14:37:03 [TRACE] (1) gitui::app: [src\app.rs:330] update
14:37:03 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:90] request: [hash: 18048908903348948695] (type: WorkingDir)
14:37:03 [TRACE] (1) asyncgit::status: [asyncgit\src\status.rs:90] request: [hash: 6368461436514963192] (type: Stage)
14:37:03 [TRACE] (4) asyncgit::status: [asyncgit\src\status.rs:143] status fetched: 6368461436514963192 (type: Stage)
14:37:03 [TRACE] (2) asyncgit::status: [asyncgit\src\status.rs:143] status fetched: 18048908903348948695 (type: WorkingDir)
14:37:03 [TRACE] (1) gitui::app: [src\app.rs:261] event: Input(Resize(118, 30))
14:37:03 [TRACE] (1) gitui::app: [src\app.rs:349] update_git: Status
14:37:03 [TRACE] (1) gitui::app: [src\app.rs:349] update_git: Status
14:37:04 [TRACE] (1) gitui::app: [src\app.rs:261] event: Input(Key(KeyEvent { code: Char('c'), modifiers: CONTROL }))
14:37:04 [ERROR] input thread error: sending on a disconnected channel
```

--- a/src/app.rs
+++ b/src/app.rs
@@ -395,8 +395,8 @@ impl App {
 	}
 
 	///
-	pub const fn is_quit(&self) -> bool {
-		self.do_quit
+	pub fn is_quit(&self) -> bool {
+		self.do_quit || self.input.is_aborted()
 	}
 
 	///

--- a/src/input.rs
+++ b/src/input.rs
@@ -3,7 +3,6 @@ use anyhow::Result;
 use crossbeam_channel::{unbounded, Receiver, Sender};
 use crossterm::event::{self, Event};
 use std::{
-	process,
 	sync::{
 		atomic::{AtomicBool, Ordering},
 		Arc,
@@ -33,6 +32,7 @@ pub struct Input {
 	desired_state: Arc<NotifyableMutex<bool>>,
 	current_state: Arc<AtomicBool>,
 	receiver: Receiver<InputEvent>,
+	aborted: Arc<AtomicBool>,
 }
 
 impl Input {
@@ -42,16 +42,18 @@ impl Input {
 
 		let desired_state = Arc::new(NotifyableMutex::new(true));
 		let current_state = Arc::new(AtomicBool::new(true));
+		let aborted = Arc::new(AtomicBool::new(false));
 
 		let arc_desired = Arc::clone(&desired_state);
 		let arc_current = Arc::clone(&current_state);
+		let arc_aborted = Arc::clone(&aborted);
 
 		thread::spawn(move || {
 			if let Err(e) =
 				Self::input_loop(&arc_desired, &arc_current, &tx)
 			{
 				log::error!("input thread error: {}", e);
-				process::abort();
+				arc_aborted.store(true, Ordering::SeqCst);
 			}
 		});
 
@@ -59,6 +61,7 @@ impl Input {
 			receiver: rx,
 			desired_state,
 			current_state,
+			aborted,
 		}
 	}
 
@@ -80,6 +83,10 @@ impl Input {
 	pub fn is_state_changing(&self) -> bool {
 		self.shall_poll()
 			!= self.current_state.load(Ordering::Relaxed)
+	}
+
+	pub fn is_aborted(&self) -> bool {
+		self.aborted.load(Ordering::SeqCst)
 	}
 
 	fn poll(dur: Duration) -> anyhow::Result<Option<Event>> {

